Assets/Scripts/Framework/Gameplay/Timer/TimerUpdater.cs

```diff
@@ -0,0 +1,135 @@
+using FishNet.Object.Synchronizing;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+using DerailedDeliveries.Framework.DamageRepairManagement;
+using DerailedDeliveries.Framework.StateMachine.States;
+using DerailedDeliveries.Framework.Station;
+using DerailedDeliveries.Framework.Utils;
+using DerailedDeliveries.Framework.Train;
+
+namespace DerailedDeliveries.Framework.Gameplay.Timer
+{
+    /// <summary>
+    /// A class that is responsible for updating the timer in the <see cref="GameState"/>.
+    /// </summary>
+    public class TimerUpdater : NetworkAbstractSingleton<TimerUpdater>
+    {
+        [SerializeField]
+        private float _baseTime = 240f;
+
+        [SerializeField]
+        private float _stationArrivalTimeBonus = 40f;
+
+        [SerializeField]
+        private float _chaosSpeedMultiplierThreshold = 60f;
+
+        [SerializeField]
+        private float _chaosSpeedMultiplier = 1.5f;
+
+        [SerializeField]
+        private List<int> _arrivedStations = new();
+
+        /// <summary>
+        /// A getter to get the ChaosSpeedMultiplierThreshold.
+        /// </summary>
+        public float ChaosSpeedMultiplierThreshold => _chaosSpeedMultiplierThreshold;
+
+        /// <summary>
+        /// A getter that returns the seconds that are left on the timer.
+        /// </summary>
+        public float TimeRemaining => _timer.Remaining;
+
+        /// <summary>
+        /// An action that broadcasts when the timer is updated and the new time that comes with it.
+        /// </summary>
+        public Action<float> OnTimerUpdated;
+
+        /// <summary>
+        /// An action that broadcasts when the timer is completed.
+        /// </summary>
+        public Action OnTimerCompleted;
```

Comment:
Nothing listens to this event, so nothing happens when the timer is finished. I think it would be nice to have a method named `EndGame()` in `GameManager.cs`.